ppingest.py

```diff
@@ -0,0 +1,96 @@
+from io import BytesIO
+from urllib.request import urlopen
+from zipfile import ZipFile
+import io
+import sys
+import csv
+import os
+from geojson.geometry import Point
```

Comment:
Unnecessary import that I missed before committing.

Reply:
Hi Joel,

I was asked to review the code by Derek.  

Compliments on the speed in which you got all this working!

Some suggestions:
1. Derek said you were using Pycharm.  IDEs are cool when you work them to your advantage.  I'd suggest you look at the this file in Pycharm.  With the file loaded there (as a .py), you'll see gray colored import statements (e.g. "import io").  Hover your mouse over that.  It'll show a yellow light bulb.  From there, click the light bulb.  It'll show a submenu: "Optimize imports".  This'll remove all unused imports in one go; otherwise use the gray coloring to show what can be removed individually.  
2. Python has a set of rules "The Zen of python" which are under the label PEP8.  For a script like this one, you wont likely get any indicators of PEP8 violations.  However, its useful to know about this helper aspect of the IDE.  When a PEP8 violation is observed it shows a green underline at the location of code.  Click on that code and Pycharm will tell you what's wrong.  It can then do things like fix format.
3. Generally, scripts are best for DevSecOps work.  It's better to write app (classes etc.), especially if its going to production.
4. One general rule for all code, is to split the content into smaller portions (for readability).  Python is very focused on readability to the coder, though other languages have other focal areas (readbility/speed for the machine).  A subset of SOLID is call Single Responsibility Principle (SRP).  It says that we should all write portions of code and classes that do one thing and one thing only.  Would advise you aim to organize code with that in mind.  Think of classes as Nouns and functions as verbs.  So read_url, extract_url,  makePoint, read_csv, insert_to_elastic, would help in readability.

I'm sure you're aware of many of these things... but neither of us are aware of each others strengths so its worth mentioning regardless.

Best to you

Evan

Reply:
Hi Mr. Fraser,

Thank you for the advice. With regards to suggestion 4, I implemented this while writing the code to ingest the power line data (using functions prepare_elasticsearch, get_file, load_file, process_features, and index_data) and it nearly tripled the amount of time it took to finish ingesting the data on my laptop. Is the increased readability worth the loss of speed? I'm considering some functions together to reduce the amount of time it takes while still keeping the code generally compartmentalized.

Thank you,
Joel Eldo